polarify/main.py

```diff
@@ -139,25 +186,104 @@ class State:
 
     node: UnresolvedState | ReturnState | ConditionalState
 
+    def translate_match(
+        self,
+        subj: ast.expr | Sequence[ast.expr] | ast.Tuple,
+        stmt: ast.pattern,
+        guard: ast.expr | None = None,
+    ):
+        if isinstance(stmt, ast.MatchValue) and isinstance(subj, ast.Name):
+            if guard is not None:
+                return ast.BinOp(
+                    left=guard,
+                    op=ast.BitAnd(),
+                    right=ast.Compare(
+                        left=ast.Name(id=subj.id, ctx=ast.Load()),
+                        ops=[ast.Eq()],
+                        comparators=[stmt.value],
+                    ),
+                )
+
+            return ast.Compare(
+                left=ast.Name(id=subj.id, ctx=ast.Load()),
+                ops=[ast.Eq()],
+                comparators=[stmt.value],
+            )
+        elif isinstance(stmt, ast.MatchValue) and isinstance(subj, ast.Tuple):
+            return self.translate_match(subj, ast.MatchSequence(patterns=[stmt]))
+        elif isinstance(stmt, ast.MatchAs) and isinstance(subj, ast.Name):
+            if stmt.name is not None:
+                self.handle_assign(
+                    ast.Assign(
+                        targets=[ast.Name(id=stmt.name, ctx=ast.Store())],
+                        value=ast.Name(id=subj.id, ctx=ast.Load()),
+                    )
+                )
+            return guard
+        elif isinstance(stmt, ast.MatchOr):
+            return ast.BinOp(
+                left=self.translate_match(subj, stmt.patterns[0], guard),
+                op=ast.BitOr(),
+                right=(
+                    self.translate_match(subj, ast.MatchOr(patterns=stmt.patterns[1:]))
+                    if stmt.patterns[2:]
+                    else self.translate_match(subj, stmt.patterns[1])
+                ),
+            )
+        elif isinstance(stmt, ast.MatchSequence):
+            if isinstance(stmt.patterns[-1], ast.MatchStar):
+                raise ValueError("starred patterns are not supported.")
+            if isinstance(subj, ast.Tuple):
+                while len(subj.elts) > len(stmt.patterns):
+                    stmt.patterns.append(ast.MatchValue(value=ast.Constant(value=None)))
+                left = self.translate_match(subj.elts[0], stmt.patterns[0], guard)
+                right = (
```

Comment:
This is not equivalent to the python behavior, see the test `match_sequence_padded_length`